### ISSUE TITLE
[feature-request] Add toggle item for "XXX server log" buffer

### ISSUE THREAD

**Issue by wztdream** [emacs-lsp/dap-mode#332]
Hi,
When debugging (python in my case) there is a "Python :: Run file xxx server log" buffer, but I never use it. So it is better to make this window customizable when dap-debug is start. 
![server](https://user-images.githubusercontent.com/39213018/89866497-38304200-dbe2-11ea-83dd-0f14df3db698.png)

Currently there is no such toggle item for this buffer in `dap-auto-configure-features`. Maybe we should add this feature in the future.
![window](https://user-images.githubusercontent.com/39213018/89866235-c821bc00-dbe1-11ea-9d48-8b534a9b2ffc.png)

**Comment by ericdallo** [comment]
I think that it's just add to the features variable the dap output buffer, am I wrong @yyoncho ?

**Comment by yyoncho** [comment]
For some adapters we do compilation-start. There is no way to manage that buffer from dap-mode ATM.

**Comment by nbfalcon** [comment]
We need to move dap-ui--show-buffer into dap-mode. There are many places where using it makes sense, and it would make dap-mode more uniform: stdio console, debugger output, ....

**Comment by yyoncho** [comment]
We should also research https://www.gnu.org/software/emacs/manual/html_node/elisp/Displaying-Buffers.html#Displaying-Buffers . I suspect that we might be able to manage the dap mode buffers using display-buffers-alist. ATM the adhoc solution dap-ui--show-buffer has some disadvantages.

**Comment by nbfalcon** [comment]
After a short look at your linked page, it seems that it is indeed possible: the action in display-buffer is the same as with display-buffer-in-side-window. This would be a breaking change though, as we couldn't continue to use display-buffer-alist + dap-ui-buffer-configurations together elegantly.

**Comment by yyoncho** [comment]
> This would be a breaking change though,

Yes. But breaking change which will result is reordering windows is something we can live with. Also, we may introduce something like `dap-display-buffer-function` which will allow reverting to the old behaviour.
